samples/apps/copilot-chat-app/webapp/src/libs/useGraph.ts

```diff
@@ -0,0 +1,158 @@
+import { useMsal } from '@azure/msal-react';
+import { Constants } from '../Constants';
+import { useAppDispatch, useAppSelector } from '../redux/app/hooks';
+import { RootState } from '../redux/app/store';
+import { addAlert } from '../redux/features/app/appSlice';
+import { UserData } from '../redux/features/users/UsersState';
+import { setUsers } from '../redux/features/users/usersSlice';
+import { TokenHelper } from './auth/TokenHelper';
+import { AlertType } from './models/AlertType';
+import { BatchRequest, BatchResponse, GraphService } from './services/GraphService';
+
+export const useGraph = () => {
+    const { instance, inProgress } = useMsal();
+    const { users } = useAppSelector((state: RootState) => state.users);
+    const { activeUserInfo } = useAppSelector((state: RootState) => state.app);
+    const dispatch = useAppDispatch();
+    const graphService = new GraphService();
+
+    const loadUsers = async (userIds: string[]) => {
+        const MAX_RETRIES = 3;
+        let retries = 1;
+
+        // Initialize empty arrays to store results
+        const usersToLoad: string[] = [];
+        const loadedUsers: UserData[] = [];
+        const usersToRetry: string[] = [];
+
+        // Copy current state of user data
+        const userData = { ...users };
+
+        try {
+            // Filter user Ids list to optimize fetch
+            userIds.forEach((userId) => {
+                const ids = userId.split('.');
+                const objectId = ids[0]; // Unique GUID assigned to each user in their home tenant
+                const tenantId = ids[1]; // Home tenant id
+
+                // Active user can only access user data within their own tenant
+                // Mark chat users outside of tenant as External
+                if (activeUserInfo && tenantId !== activeUserInfo.id.split('.')[1]) {
```

Comment:
nit: can we save this call to a variable so we don't have to do the .split() each time?